### PR TITLE
Include charge reweighting for Phase-2 IT planar sensors

### PR DESCRIPTION
**Pull request by tvami**
#### PR description:

Follow-up to https://github.com/cms-sw/cmssw/pull/40349
Add the boolean `UseReweighting` to Phase-2 Tracker digitizers in order to simulate charge loss due to irradiation. This uses the same (now generalized) method that is use in the Phase-0/1 simulations. For now it's set to False by default. For now I didn't touch the late charge reweighting, I believe it's not (yet) relevant for Phase-2

#### PR validation:

`20834.0` ran fine with the default `False` value.
After the compilation issue for Phase-1, I also ran `250202.184` which tests the late charge rew (this failed with some input error), and `11634.0` which is for realistic 2021.

when `UseReweighting = True`, I used the `20834.0`  wf but modified the step2 to 
```
cmsDriver.py step2 -s DIGI:pdigi_valid,L1TrackTrigger,L1,DIGI2RAW,HLT:@fake2 --conditions 125X_mcRun4_realistic_Candidate_2023_01_03_23_47_45 --datatier GEN-SIM-DIGI-RAW -n 10 --eventcontent FEVTDEBUGHLT --geometry Extended2026D88 --era Phase2C17I13M9 --filein file:step1.root --fileout file:step2.root
```
where `125X_mcRun4_realistic_Candidate_2023_01_03_23_47_45` contains the 2D templates needed for the charge reweighting. This runs fine as well.

Run LA trees and showed the charge vs depth profiles with the `True` option, the outcome looks as expected.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport and no backport is needed.

### CODE REVIEW

**Comment by tvami** [comment]
test parameters:
  - workflows = 20834.0,24434.0

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h [hunk 1]
```diff
@@ -143,51 +142,54 @@ inline SiPixelChargeReweightingAlgorithm::SiPixelChargeReweightingAlgorithm(cons
       UseReweighting(conf.getParameter<bool>("UseReweighting")),
       applyLateReweighting_(conf.exists("applyLateReweighting") ? conf.getParameter<bool>("applyLateReweighting")
                                                                 : false),
-      PrintClusters(conf.getParameter<bool>("PrintClusters")),
-      PrintTemplates(conf.getParameter<bool>("PrintTemplates")) {
+      PrintClusters(conf.exists("PrintClusters") ? conf.getParameter<bool>("PrintClusters") : false),
```
Comment (mmusich) [inline SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:145]: this practice is discouraged. Time to add a `fillDescriptions`.

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h [hunk 2]
```diff
@@ -143,51 +142,54 @@ inline SiPixelChargeReweightingAlgorithm::SiPixelChargeReweightingAlgorithm(cons
...
+      PrintTemplates(conf.exists("PrintTemplates") ? conf.getParameter<bool>("PrintTemplates") : false) {
```
Comment (mmusich) [inline SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:146]: this practice is discouraged. Time to add a `fillDescriptions`.

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h [hunk 3]
```diff
@@ -669,7 +672,14 @@ inline bool SiPixelChargeReweightingAlgorithm::lateSignalReweight(const PixelGeo
     unsigned int chan = loopDigi->channel();
     if (loopTempSH.isInTheList(chan)) {
       float corresponding_charge = loopDigi->adc();
-      theDigiSignal[chan] += digitizerUtility::Amplitude(corresponding_charge, corresponding_charge);
+      if constexpr (std::is_same_v<AmplitudeType, digitizerUtility::Ph2Amplitude>) {
+        edm::LogError("SiPixelChargeReweightingAlgorithm")
+            << "Phase-2 late charge reweighting not supported (not sure we need it at all)";
+        return false;
+        //theDigiSignal[chan] += AmplitudeType(corresponding_charge, &hit, corresponding_charge, 0., hitIndex, tofBin);
```
Comment (mmusich) [inline SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:679]: commented out code (after a `return` statement)

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h [hunk 4]
```diff
@@ -794,8 +802,16 @@ inline bool SiPixelChargeReweightingAlgorithm::lateSignalReweight(const PixelGeo
       if ((hitPixel.first + row - THX) >= 0 && (hitPixel.first + row - THX) < topol->nrows() &&
           (hitPixel.second + col - THY) >= 0 && (hitPixel.second + col - THY) < topol->ncolumns() && charge > 0) {
         chargeAfter += charge;
-        theNewDigiSignal[PixelDigi::pixelToChannel(hitPixel.first + row - THX, hitPixel.second + col - THY)] +=
-            digitizerUtility::Amplitude(charge, charge);
+        if constexpr (std::is_same_v<AmplitudeType, digitizerUtility::Ph2Amplitude>) {
+          edm::LogError("SiPixelChargeReweightingAlgorithm")
+              << "Phase-2 late charge reweighting not supported (not sure we need it at all)";
+          //theNewDigiSignal[PixelDigi::pixelToChannel(hitPixel.first + row - THX, hitPixel.second + col - THY)] +=
+          //    AmplitudeType(charge, &hit, charge, 0., hitIndex, tofBin);
```
Comment (mmusich) [inline SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:809]: commented code.

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h [hunk 5]
```diff
@@ -821,7 +840,14 @@ inline bool SiPixelChargeReweightingAlgorithm::lateSignalReweight(const PixelGeo
     if (i_transformed_row < 0 || i_transformed_row > TXSIZE || i_transformed_col < 0 || i_transformed_col > TYSIZE) {
       // not in the box
       if (chanDigi >= 0 && (*i).second > 0) {
-        theNewDigiSignal[chanDigi] += digitizerUtility::Amplitude((*i).second, (*i).second);
+        if constexpr (std::is_same_v<AmplitudeType, digitizerUtility::Ph2Amplitude>) {
+          edm::LogError("SiPixelChargeReweightingAlgorithm")
+              << "Phase-2 late charge reweighting not supported (not sure we need it at all)";
+          return false;
+          //theNewDigiSignal[chanDigi] += AmplitudeType((*i).second, &hit, (*i).second, 0., hitIndex, tofBin);
```
Comment (mmusich) [inline SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:847]: commented code.

File: SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc [hunk 6]
```diff
@@ -563,14 +567,51 @@ void Phase2TrackerDigitizerAlgorithm::induce_signal(
     }
   }
   // Fill the global map with all hit pixels from this event
+  bool reweighted = false;
+  size_t ReferenceIndex4CR = 0;
```
Comment (mmusich) [inline SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc:571]: does it have to be upper case? It appears it does not directly contradict `cmssw` code rules, but looks unusual.

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h [hunk 7]
```diff
@@ -805,12 +821,15 @@ inline bool SiPixelChargeReweightingAlgorithm::lateSignalReweight(const PixelGeo
...
-    LogDebug("Pixel Digitizer") << "Charge loss: " << (1 - chargeAfter / chargeBefore) * 100 << " %";
+    LogDebug("SiPixelChargeReweightingAlgorithm")
+        << "Charges (before->after): " << chargeBefore << " -> " << chargeAfter;
+    LogDebug("SiPixelChargeReweightingAlgorithm") << "Charge loss: " << (1 - chargeAfter / chargeBefore) * 100 << " %";
   }
 
   // need to store the digi out of the 21x13 box.
-  for (signal_map_const_iterator i = theDigiSignal.begin(); i != theDigiSignal.end(); ++i) {
+  for (std::map<int, digitizerUtility::Amplitude, std::less<int> >::const_iterator i = theDigiSignal.begin();
+       i != theDigiSignal.end();
```
Comment (mmusich) [inline SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:831]: range based?

File: SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py [hunk 8]
```diff
@@ -227,27 +234,32 @@
         AddNoisyPixels = False,
         AddInefficiency = False,
         AddThresholdSmearing = False,
+        UseReweighting = False,
```
Comment (mmusich) [inline SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py:237]: not sure to understand why you want to do that (in the case of the late re-weighting).
In any case please add a line of comment explaining why it is done after line 224 (in the proposed branch).

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h [hunk 9]
```diff
@@ -3,8 +3,8 @@
 
 // Original Author Caroline Collard
 // September 2020 : Extraction of the code for cluster charge reweighting from SiPixelDigitizerAlgorithm to a new class
+// Jan 2023: Generalize so it can be used for Phase-2 IT as well (Tamas Vami)
```
Comment (mmusich) [inline SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:6]: the overall approach (including https://github.com/cms-sw/cmssw/pull/40349) looks reminescent of https://github.com/mmusich/cmssw/commit/f1b1ff5453edc1125a35fdc4f73040fa06385cd2

**Review by mmusich — CHANGES_REQUESTED** [review]

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h (hunk 9 above)
Comment (tvami) [reply SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:6]: yes, I'll add your name, thanks!

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h (hunk 1 above)
Comment (tvami) [reply SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:145]: I know, and I agree with you that it's time to add a `fillDescriptions`, I was even thinking about doing it... but I'm not the official maintainer of these codes... the digitizer (both Phase-1 and Phase-2) is lacking the `fillDescriptions` too. So maybe I can do it, but then I'd prefer to do it as a campaign for all these in a separate PR

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h (hunk 3 above)
Comment (tvami) [reply SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:679]: will do in the next push

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h (hunk 4 above)
Comment (tvami) [reply SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:809]: will do in the next push

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h (hunk 5 above)
Comment (tvami) [reply SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:847]: will do in the next push

File: SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc (hunk 6 above)
Comment (tvami) [reply SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc:571]: sure I can change it

**Comment by tvami** [comment]
I see a few widespread changes for the Phase-2 wf:
https://tinyurl.com/2zu8vzpk

is this just because we redigitized?
Can the error change due to that? https://tinyurl.com/2jc5puk7 the target looks better than then ref... so maybe this is coming from another PR?

**Comment by tvami** [comment]
OK I tested the version where the Charge Rew is True, things look as expected, so I'm moving the PR away from being a draft

**Comment by tvami** [comment]
@cmsbuild , please test

File: SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py (hunk 8 above)
Comment (mmusich) [reply SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py:237]: This block is needed to feed a modifier. If that's the default behaviour, change it in the default configuration. If it's instead needed only for the premix stage1, please document it in the comments as already requested.

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h (hunk 1 above)
Comment (mmusich) [reply SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:145]: Since these switches are seemingly used for debug and hence should not change physics, I suppose you could make them `untracked` and provide default values. In a second round let's go with the `fillDescriptions`.

**Comment by mmusich** [comment]
> OK I tested the version where the Charge Rew is True, things look as expected

I understand this will be shown at the next Phase2 Tracker  Simulation meeting (Jan 20th). I supposed it would be good to add a reference to it in the PR description, once it's available.

File: SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h (hunk 1 above)
Comment (tvami) [reply SimTracker/Common/interface/SiPixelChargeReweightingAlgorithm.h:145]: ok I fixed these, and will do the `fillDescriptions` next time

File: SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py (hunk 8 above)
Comment (tvami) [reply SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py:237]: Where is that supposed to be? I tried in`SiPixelSimBlock` part but that still tells me that the mixing is missing `UseReweighting`

**Comment by mmusich** [comment]
@tvami 

> I see widespread changes for the Phase-2 wf:
https://tinyurl.com/2zu8vzpk
> is this just because we redigitized?
> Can the error change due to that? https://tinyurl.com/2jc5puk7 the target looks better than then ref... so maybe this is coming from another PR?

as far as I can see, the widespread changes in wf [24434.0](https://cmsweb.cern.ch/dqm/dev/start?runnr=1;dataset%3D/RelVal_wf24434_0_pr/CMSSW_13_0_X-PRcmssw_40417-54835/DQMIO;referenceshow%3Dall;referencenorm=False;referenceobj1%3Dother::/RelVal_wf24434_0_base/CMSSW_13_0_X-PRcmssw_40417-54835/DQMIO::;sampletype%3Doffline_relval;workspace%3DEverything;) persist also in the [last round of tests](https://github.com/cms-sw/cmssw/pull/40417#issuecomment-1373008956).
They do NOT seem to go in the right direction, in particular:
   * the amount of missed digis in BPix 1 is greatly increased (previously was zero, cf https://tinyurl.com/2j5q3pzm)
   * there are NO `trackingRechits` in BPix 1 (see https://tinyurl.com/2hdt5eqe)
   * there is a large loss of tracks seeded by `initialStep` (high pT quadruplets) while there is a large increase of tracks seeded by the `highPTTriplet` step (which is a recovery iteration): https://tinyurl.com/2hjcdw4l 

The circumstantial evidence is that the BPix 1 `SimHit`s are not digitized at all in this workflow (which employs geometry D97, i.e. Tracker T25, see [README](https://github.com/cms-sw/cmssw/blob/master/Configuration/Geometry/README.md)). Given that the sister workflow 23234.0 (employing geometry D94, i.e. Tracker T24) exhibits [no changes at all](https://tinyurl.com/2eph5c58) (as expected), I am lead to conclude that your changes are breaking the digitization for the 3D Pixels.

File: SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.cc [hunk 10]
```diff
@@ -343,11 +336,12 @@ std::vector<digitizerUtility::SignalPoint> Pixel3DDigitizerAlgorithm::drift(
 // Signal is already "induced" (actually electrons transported to the
 // n-column) at the electrode. Just collecting and adding-up all pixel
 // signal and linking it to the simulated energy deposit (hit)
-void Pixel3DDigitizerAlgorithm::induce_signal(const PSimHit& hit,
-                                              const size_t hitIndex,
-                                              const uint32_t tofBin,
-                                              const Phase2TrackerGeomDetUnit* pixdet,
-                                              const std::vector<digitizerUtility::SignalPoint>& collection_points) {
+void Pixel3DDigitizerAlgorithm::induce_signal3D(std::vector<PSimHit>::const_iterator inputBegin,
```
Comment (mmusich) [inline SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.cc:339]: why is this changed of name? I thought this was meant to override the method of the mother class:

https://github.com/cms-sw/cmssw/blob/44c3f64c0f5ce5c08f31fb622bc94de026c84263/SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.h#L193-L197

in this way this method is never called.
@duartej for your information.

File: SimTracker/SiPhase2Digitizer/plugins/PixelBrickedDigitizerAlgorithm.cc [hunk 11]
```diff
@@ -50,11 +50,12 @@ PixelBrickedDigitizerAlgorithm::PixelBrickedDigitizerAlgorithm(const edm::Parame
 PixelBrickedDigitizerAlgorithm::~PixelBrickedDigitizerAlgorithm() {
   LogDebug("PixelBrickedDigitizerAlgorithm") << "Algorithm deleted";
 }
-void PixelBrickedDigitizerAlgorithm::induce_signal(const PSimHit& hit,
-                                                   const size_t hitIndex,
-                                                   const uint32_t tofBin,
-                                                   const Phase2TrackerGeomDetUnit* pixdet,
-                                                   const std::vector<digitizerUtility::SignalPoint>& collection_points) {
+void PixelBrickedDigitizerAlgorithm::induce_signalBricked(
```
Comment (mmusich) [inline SimTracker/SiPhase2Digitizer/plugins/PixelBrickedDigitizerAlgorithm.cc:53]: why is this changed of name? I thought this was meant to override the method of the mother class:

https://github.com/cms-sw/cmssw/blob/44c3f64c0f5ce5c08f31fb622bc94de026c84263/SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.h#L193-L197

in this way this method is never called.

File: SimTracker/SiPhase2Digitizer/plugins/PixelBrickedDigitizerAlgorithm.h [hunk 12]
```diff
@@ -17,10 +17,10 @@ class PixelBrickedDigitizerAlgorithm : public PixelDigitizerAlgorithm {
   ~PixelBrickedDigitizerAlgorithm() override;
 
   // Specific for bricked pixel
-  void induce_signal(const PSimHit& hit,
-                     const size_t hitIndex,
-                     const unsigned int tofBin,
-                     const Phase2TrackerGeomDetUnit* pixdet,
-                     const std::vector<digitizerUtility::SignalPoint>& collection_points) override;
+  void induce_signalBricked(const PSimHit& hit,
```
Comment (mmusich) [inline SimTracker/SiPhase2Digitizer/plugins/PixelBrickedDigitizerAlgorithm.h:20]: please revert.

File: SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.h [hunk 13]
```diff
@@ -52,11 +52,12 @@ class Pixel3DDigitizerAlgorithm : public PixelDigitizerAlgorithm {
                                                              const std::pair<float, float> pitches,
                                                              const float& thickness) const;
   // Specific for 3D-pixel
-  void induce_signal(const PSimHit& hit,
-                     const size_t hitIndex,
-                     const uint32_t tofBin,
-                     const Phase2TrackerGeomDetUnit* pixdet,
-                     const std::vector<digitizerUtility::SignalPoint>& collection_points) override;
+  void induce_signal3D(std::vector<PSimHit>::const_iterator inputBegin,
```
Comment (mmusich) [inline SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.h:55]: please revert.

**Review by mmusich — CHANGES_REQUESTED** [review]
I think the issue with the 3D digitization comes from the fact that you altered the names of the `induce_signal` methods that are meant to shadow the one of the mother class.

File: SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.cc (hunk 10 above)
Comment (tvami) [reply SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.cc:339]: I changed the name because I got this msg
"Pixel3DDigitizerAlgorithm.h:55:8: error: 'void Pixel3DDigitizerAlgorithm::induce_signal()' marked 'override', but does not override"
but I guess that was the wrong resolution of the problem, I should have just made it `virtual`

Comment (mmusich) [reply SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.cc:339]: > I should have just made it `virtual`

I fail to see how that would solve the issue. The argument list of the derived class method should match the one of the mother class to allow the shadowing of the virtual mother class method from the concrete implementation.

Comment (tvami) [reply SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.cc:339]: So making it virtual gives 
```
hidden overloaded virtual function 'Phase2TrackerDigitizerAlgorithm::induce_signal' declared here: different number of parameters (7 vs 6)
```

File: SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py (hunk 8 above)
Comment (mmusich) [reply SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py:237]: whic workflow did you use for testing?

Comment (tvami) [reply SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py:237]: I used `20834.0`

Comment (mmusich) [reply SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py:237]: I am confused.
I tested this patch on top of your branch:
```diff
diff --git a/SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py b/SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py
index 47a0060e2c5..d9ec3dc3264 100644
--- a/SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py
+++ b/SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py
@@ -234,32 +234,32 @@ _premixStage1ModifyDict = dict(
         AddNoisyPixels = False,
         AddInefficiency = False,
         AddThresholdSmearing = False,
-        UseReweighting = False,
+        #UseReweighting = False,
     ),
     Pixel3DDigitizerAlgorithm = dict(
         AddNoisyPixels = False,
         AddInefficiency = False,
         AddThresholdSmearing = False,
-        UseReweighting = False,
+        #UseReweighting = False,
     ),
     PSPDigitizerAlgorithm = dict(
         AddNoisyPixels = False,
         AddInefficiency = False,
         AddThresholdSmearing = False,
-        UseReweighting = False,
+        #UseReweighting = False,
     ),
     PSSDigitizerAlgorithm = dict(
         AddNoisyPixels = False,
         AddInefficiency = False,
         AddThresholdSmearing = False,
         Phase2ReadoutMode = -1,
-        UseReweighting = False,
+        #UseReweighting = False,
     ),
     SSDigitizerAlgorithm = dict(
         AddNoisyPixels = False,
         AddInefficiency = False,
         AddThresholdSmearing = False,
-        UseReweighting = False,
+        #UseReweighting = False,
     ),
 )
```
and `runTheMatrix.py -l 20834.0 -t 4 -j 8 --ibeos` runs without problems for me.
Do I miss something?

Comment (tvami) [reply SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py:237]: Yeah it indeed work, I guess this msg I referred to was at an earlier step in the development

**Comment by tvami** [comment]
@mmusich I think I resolved all your comments, are we ready to test?

**Comment by mmusich** [comment]
> are we ready to test?

sure.

**Review by mmusich — APPROVED** [review]

**Comment by mmusich** [comment]
> DQMHistoTests: Total failures: 160

Comparisons at https://github.com/cms-sw/cmssw/pull/40417#issuecomment-1376170683 now look good. Differences are either unrelated or spurious (linked to the known problematic wf 11634.7, see https://github.com/cms-sw/cmssw/issues/39803).
On the other hand we might want to re-trigger tests to have a cleaner picture.

**Comment by tvami** [comment]
> Reco comparison results: 0 differences found in the comparisons

Yay!
@cms-sw/simulation-l2 @cms-sw/upgrade-l2 do you have any comments?

**Comment by mmusich** [comment]
type trk, new-feature

**Comment by mmusich** [comment]
new-feature

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

New feature as explained in the PR description. The default is still not to use it, so no change is expected.

**Comment by perrotta** [comment]
+1
